tests/cancun/eip1153_tstore/test_tstorage.py

```diff
@@ -246,3 +246,52 @@ def test_gas_usage(
         TestAddress: Account(nonce=1),
     }
     state_test(env=env, pre=pre, tx=tx, post=post)
+
+
+@unique
+class LoopRunUntilOutOfGasCases(PytestParameterEnum):
+    """
+    Test cases to run until out of gas.
+    """
+
+    TSTORE = {
+        "description": "Run tstore in loop until out of gas",
+        "repeat_bytecode": Op.TSTORE(Op.GAS, Op.GAS),
+        "bytecode_repeat_times": 1000,
+    }
+    TSTORE_WIDE_ADDRESS_SPACE = {
+        "description": "Run tstore in loop until out of gas, using a wide address space",
+        "repeat_bytecode": Op.TSTORE(Op.ADD(Op.SHL(Op.PC, 1), Op.GAS), Op.GAS),
+        "bytecode_repeat_times": 32,
+    }
+    TSTORE_TLOAD = {
+        "description": "Run tstore and tload in loop until out of gas",
+        "repeat_bytecode": Op.GAS + Op.DUP1 + Op.DUP1 + Op.TSTORE + Op.TLOAD + Op.POP,
+        "bytecode_repeat_times": 1000,
+    }
+
+
+@LoopRunUntilOutOfGasCases.parametrize()
+def test_run_until_out_of_gas(
+    state_test: StateTestFiller,
+    repeat_bytecode: bytes,
+    bytecode_repeat_times: int,
+):
+    """
+    Use TSTORE over and over to different keys until we run out of gas.
+    """
+    bytecode = Op.JUMPDEST + repeat_bytecode * bytecode_repeat_times + Op.JUMP(Op.PUSH0)
```

Comment:
Would there be any value in adding an `Op.SSTORE(0, 1)` to the end of the byte code and checking that the contract's storage is still 0 in `post`?

Reply:
It is better to have oog in subcall
So that subcall runs oog and reverts
But the main call still puts
Sstor(code_worked, 1)